### PR TITLE
feat: Rseqc inner distance

### PR DESCRIPTION
**Pull request by tdayris**
### QC

* [x] I confirm that I have followed the [documentation for contributing to `snakemake-wrappers`](https://snakemake-wrappers.readthedocs.io/en/stable/contributing.html).

While the contributions guidelines are more extensive, please particularly ensure that:
* [x] `test.py` was updated to call any added or updated example rules in a `Snakefile`
* [x] `input:` and `output:` file paths in the rules can be chosen arbitrarily
* [x] wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`)
* [x] temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to 
* [x] the `meta.yaml` contains a link to the documentation of the respective tool or command under `url:`
* [x] conda environments use a minimal amount of channels and packages, in recommended ordering

## Summary by CodeRabbit

- **New Features**
  - Introduced `environment.linux-64.pin.txt` for a reproducible Conda environment setup on Linux 64-bit.
  - Added `environment.yaml` to streamline the installation of the `rseqc` package and its dependencies.
  - Launched the `inner_distance` tool to calculate inner distances between read pairs in genomic data.
  - Implemented a structured `annotation.bed` file for genomic feature representation.

- **Enhancements**
  - Added a Snakemake rule for analyzing inner distances in RNA sequencing data.
  - Created a wrapper to facilitate the integration of the inner distance functionality into Snakemake workflows.

- **Tests**
  - Included a new test case for the inner distance calculation process within the bioinformatics toolkit.

### CODE REVIEW

File: bio/rseqc/inner_distance/test/Snakefile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+rule test_rseqc_inner_distance:
+    input:
+        aln="A.bam",
+        refgene="annotation.bed",
+    output:
+        txt="a.txt",
```
Comment (fgvieira) [inline bio/rseqc/inner_distance/test/Snakefile:6]: Maybe use a more specific name than `txt`. What does this file contain?

File: test.py [hunk 2]
```diff
@@ -6474,6 +6474,14 @@ def test_generate_data_matrix():
     )
 
 
+@skip_if_not_modified
+def test_rseqc_inner_distance():
+    run(
+        "bio/rseqc/inner_distance",
+        ["snakemake", "--cores", "1", "--use-conda", "-F"],
```
Comment (fgvieira) [inline test.py:6481]: Maybe be a bit more specific:
```suggestion
        ["snakemake", "--cores", "1", "--use-conda", "-F", "a.pdf"],
```

File: bio/rseqc/inner_distance/test/Snakefile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        freq="feqt.txt",
```
Comment (fgvieira) [inline bio/rseqc/inner_distance/test/Snakefile:7]:
```suggestion
        freq="freq.txt",
```

File: bio/rseqc/inner_distance/wrapper.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+__author__ = "Thibault Dayris"
+__mail__ = "thibault.dayris@gustaveroussy.fr"
+__copyright__ = "Copyright 2024, Thibault Dayris"
+__license__ = "MIT"
+
+from tempfile import TemporaryDirectory
+from snakemake import shell
+
+extra = snakemake.params.get("extra", "")
+log = snakemake.log_fmt_shell(stdout=True, stderr=True, append=False)
```
Comment (fgvieira) [inline bio/rseqc/inner_distance/wrapper.py:14]: If not appending, the logs will be overwritten, no?

File: bio/rseqc/inner_distance/test/Snakefile (hunk 1 above)
Comment (tdayris) [reply bio/rseqc/inner_distance/test/Snakefile:6]: It's a trash-file IMO. I don't know any tool that uses this file (multiqc uses the `output.freq` file, both PDF and R scripts are used by us, humans). RSeQC itself never reads the content of this file.

This file is a TSV is not described by any official documentation. I just read the code, and, for each read in the input (bam/sam), it contains the following: read name <tab> inner_distance if no filtered else NA <tab> comments about read pair.

Let's call it `reads_inner_distance` ?

**Review by fgvieira — APPROVED** [review]

**Pull request merged**